.github/workflows/lint.yml

```diff
@@ -0,0 +1,82 @@
+name: lint
```

Comment:
Separating lint task in their own workflow, and enabling lint for windows and freebsd

Reply:
This PR is still too large to review.
Can you split the GHA part to a separate PR?

Reply:
Even with all the separate commits? (with 7 commits, that should be about only 100 lines per commit)

I can break in different _PRs_ of course if this is better for you, but rebase-ing interactively across branches is not particularly pleasant...

Reply:
Yes, that would be appreciated as I can make a partial approval/rejection.

I know rebasing is painful, but the most of the GHA part seems independent from the Go part and probably doesn’t need rebasing